src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/IndexerCompletionProvider.cs

```diff
@@ -34,7 +34,7 @@ public IndexerCompletionProvider()
 
         protected override ImmutableArray<CompletionItem> GetCompletionItemsForTypeSymbol(ITypeSymbol container, bool isAccessedByConditionalAccess, ExpressionSyntax expression, SemanticModel semanticModel, int position, CancellationToken cancellationToken)
         {
-            var expressionSymbolContainingType = semanticModel.GetSymbolInfo(expression, cancellationToken).Symbol?.GetContainingTypeOrThis();
+            var expressionSymbolContainingType = (semanticModel.GetSymbolInfo(expression, cancellationToken).Symbol ?? semanticModel.GetEnclosingSymbol(position, cancellationToken))?.GetContainingTypeOrThis();
```

Comment:
1. introduce local.
2. explain why you're doing it this way :)

Reply:
> explain why you're doing it this way :)

Well, I think the best explanation I can give that it was close to midnight when I wrote it 😅 .

I simplified it to
```CS
            // We only want to suggest indexer accessible from the cursor position, so we need the containing type at the cursor position,
            // because the within parameter of GetAccessibleMembersInThisAndBaseTypes() must be an IAssemblySymbol or an INamedTypeSymbol.
            var containingTypeAtCursorPosition = semanticModel.GetEnclosingSymbol(position, cancellationToken)?.GetContainingTypeOrThis();
```
I hope this is accurate now.